src/main/java/edu/wpi/grip/ui/pipeline/input/InputSocketViewFactory.java

```
@@ -0,0 +1,64 @@
+        final SocketHint<T> socketHint = socket.getSocketHint();
+
+        switch (socketHint.getView()) {
+            case NONE:
+                return new InputSocketView<T>(eventBus, socket) {
+                };
+
+            case SPINNER:
+                if (socketHint.getType().equals(Number.class)) {
+                    return (InputSocketView<T>) new NumberSpinnerInputSocketView(eventBus, (InputSocket<Number>) socket);
+                } else {
+                    return new ListSpinnerInputSocketView<>(eventBus, socket);
+                }
+
+            case SLIDER:
+                if (socketHint.getType().equals(Number.class)) {
+                    return (InputSocketView<T>) new SliderInputSocketView(eventBus, (InputSocket<Number>) socket);
+                } else {
+                    throw new RuntimeException("Could not create view for socket.  SLIDER views must be Numbers. "
+                            + socket.toString());
+                }
+
+            case RANGE:
+                if (socketHint.getType().equals(List.class)) {
+                    return (InputSocketView<T>) new RangeInputSocketView(eventBus, (InputSocket<List<Number>>) socket);
+                } else {
+                    throw new RuntimeException("Could not create view for socket.  RANGE views must be Lists. "
+                            + socket.toString());
+                }
+
+
+            case SELECT:
+                return new SelectInputSocketView<>(eventBus, socket);
+
+            case CHECKBOX:
+                if (socketHint.getType().equals(Boolean.class)) {
+                    return (InputSocketView<T>) new CheckboxInputSocketView(eventBus, (InputSocket<Boolean>) socket);
+                } else {
+                    throw new RuntimeException("Could not create view for socket.  CHECKBOX views must be Booleans. "
+                            + socket.toString());
+                }
```

Yes? But there should be a default case in a case switch statement. That final default case should throw the exception not the way it is now. Sorry, I seem to have put this comment on the wrong line.